src/rust/bitbox02-rust/src/hww.rs

```diff
@@ -25,6 +25,25 @@ const OP_STATUS_SUCCESS: u8 = 0;
 const OP_STATUS_FAILURE: u8 = 1;
 const OP_STATUS_FAILURE_UNINITIALIZED: u8 = 2;
 
+/// Must be called during the execution of a usb task. This sends out the response to the host and
+/// awaits the next request. If the request is not a valid noise encrypted protofbuf api request
+/// message, `Err(Error::InvalidInput)` is returned.
+pub async fn next_request(
+    response: crate::pb::response::Response,
+) -> Result<crate::pb::request::Request, api::error::Error> {
+    let mut out = [OP_STATUS_SUCCESS].to_vec();
+    noise::encrypt(&api::encode(response), &mut out).or(Err(api::error::Error::Generic))?;
```

Comment:
I'd rather make an Error::Noise. It probably won't happen but if it does, easy to identify where it's coming from, especially in the future if there's a bug in the noise bits.

Reply:
Also below at noise::decrypt. How about `Error:NoiseEncrypt` and `Error::NoiseDecrypt`?

Reply:
Added, PTAL. See also the extended commit message:

> The new noise errors are likely never delivered to the host
successfully. If encryption fails, the error response itself also will
likely fail to be encrypted. If decryption fails, the host will likely
not be able to decrypt the encrypted error response either.

Any errors here are likely not recoverable and a replug will be needed.

Reply:
Yeah for sure but I still think it's a good practice. Maybe in the future there will be an abort showing the error on the display or something like that. Helps maintenance and debugging imho.